tests/rptest/redpanda_cloud_tests/high_throughput_test.py

```diff
@@ -468,12 +468,26 @@ def _get_swarm_connections_count(self, nodes):
                 _list.append(_current)
         return _list
 
+    def _get_hwm_for_default_topic(self):
+        _hwm = 0
+        for partition in self.rpk.describe_topic(self.topic):
+            # Add currect high watermark for topic
+            _hwm += partition.high_watermark
+        return _hwm
+
     @cluster(num_nodes=9)
     def test_max_connections(self):
```

Comment:
nit: would be nice to have a pydoc comment that would describe the strategy of the test for validating max connections:
1. create client swarm connections up to the max advertised for the tier
2. count the messages sent and ensure at least a sizeable amount has gone through because just having a network connection is not solid enough to ensure data is flowing on the wire